src/slashml.py

```diff
@@ -0,0 +1,85 @@
+import requests
+import os
+from pathlib import Path
+import json
+
+class SpeechToText:
+
+    SLASHML_BASE_URL = 'https://api.slashml.com/speech-to-text/v1'
+    SLASHML_UPLOAD_URL = SLASHML_BASE_URL+'/upload/'
+    SLASHML_TRANSCRIPT_URL = SLASHML_BASE_URL+'/transcribe/'
+    SLASHML_STATUS_URL = SLASHML_BASE_URL+'/transcription'
+    SLASHML_TRANSCRIPT_STATUS_URL = lambda self,id: f"{SpeechToText.SLASHML_STATUS_URL}/{id}/"
+
+    HEADERS:dict = {}
+    ## add the api key to sys path envs 
+    def __init__(self,API_KEY: str = None):
+        self.API_KEY=None
+        if API_KEY:
+            token="Token {0}".format(API_KEY)
+            self.HEADERS = {'authorization': token}
+           # print("Auth with "+API_KEY+"\nMake sure this matches your API Key generated in the dashboard settings")
+        elif os.environ.get('SLASHML_API_KEY'):
+            key_env =  os.environ.get('SLASHML_API_KEY')
+            token="Token {0}".format(key_env)
+            self.HEADERS = {'authorization': token}
+           # print("Auth with environment variable SLASHML_API_KEY")
+        else:
+            self.HEADERS=None 
+            print("No Auth, there are certain limites to the usage")
+
+    def upload_audio(self, file_location:str,header=None):
+        headers = self.HEADERS
+        files=[
+        ('audio',('test_audio.mp3',open(file_location,'rb'),'audio/mpeg'))
+        ]
+        response = requests.post(self.SLASHML_UPLOAD_URL,
+                                headers=headers,files=files)
+        return  response.json()["upload_url"]
```

Comment:
there is a possibility that you receive 400 or 500 when uploading the file. When that happens the response.json() might not contain `upload_url`.

Reply:
i added conditions on this, if the response is success get the upload URL.